File: integrations/src/main/java/com/nivasafinance/integrations/framework/core/logger/ThirdPartyRequestResponseLogger.java
package com.nivasafinance.integrations.framework.core.logger;

import com.nivasafinance.integrations.framework.config.BusinessContext;
import com.nivasafinance.integrations.framework.core.data.ApiContext;
import org.springframework.http.HttpMethod;


public interface ThirdPartyRequestResponseLogger {

    Long registerRequest(
            BusinessContext businessContext,
            ApiContext apiContext,
            HttpMethod requestMethod,
            String url,
            String requestBody
    );

    Long registerThirdPartyRequest(
            BusinessContext businessContext,
            ApiContext apiContext,
            HttpMethod requestMethod,
            String url,
            String requestBody
    );

    void registerResponse(
            Long thirdPartyRequestId,
            String responseBody,
            long responseTimeInMs,
            int requestStatus
    );
}

File: integrations/src/main/kotlin/framework/core/logger/ThirdPartyRequestResponseLogger.kt
package framework.core.logger

import framework.config.BusinessContext
import framework.core.data.ApiContext
import org.springframework.http.HttpMethod
import java.util.UUID

interface ThirdPartyRequestResponseLogger {

    fun registerRequest(
        businessContext: BusinessContext,
        apiContext: ApiContext,
        requestMethod: HttpMethod,
        url: String,
        requestBody: String?
    ): UUID

    fun registerThirdPartyRequest(
        businessContext: BusinessContext,
        apiContext: ApiContext,
        requestMethod: HttpMethod,
        url: String,
        requestBody: String?
    ): UUID

    fun registerResponse(
        thirdPartyRequestId: UUID,
        responseBody: String,
        responseTimeInMs: Long,
        requestStatus: Int
    )
}
